schematics/models.py

```diff
@@ -165,12 +165,16 @@ def __new__(mcs, name, bases, attrs):
 
         # Add reference to klass to each field instance
         def set_owner_model(field, klass):
-            field.owner_model = klass
-            if hasattr(field, 'field'):
+            try:
+                field.owner_model = klass
                 set_owner_model(field.field, klass)
+            except AttributeError:
```

Comment:
What do you mean? I'm using a specific `AttributeError` exception. I can write a regression test for this if you want.